website/api/zkvm/input-output.md

```diff
@@ -0,0 +1,274 @@
+# Introduction to Input/Output in the zkVM
+
+This document provides an overview of how to perform I/O operations in the zkVM,
+going over the different file descriptors and methods used to handle inputs and
+outputs, the distinction between public and private data and additional
+considerations when working with I/O in the zkVM.
+
+As briefly mentioned in [Guest Code 101][guest-code-101-io], we can perform I/O
+between the guest and host by using some methods under the [`env`
+module][env-module-functions]. These methods allow sending private data between
+them. In the zkVM, there is a distinction between public and private data.
+Public data is included in the journal and is part of the proof, while private
+data is sent between the guest and host and is not included in the journal nor
```

Comment:
we need to both be careful about creating barriers by assuming familiarity with jargon *and* be careful of explaining everything on every page. my preference is to keep things succinct and link out to more info as appropriate. 

i..e., here, i'd suggest we just link to the terminology pages for guest and host rather than explaining the terms in-line